presto-yarn-test/src/test/groovy/com/teradata/presto/yarn/PrestoClusterTest.groovy

```diff
@@ -66,6 +67,31 @@ class PrestoClusterTest
   @Named("cluster.slaves")
   private List<String> workers;
 
+  @BeforeTestWithContext
+  public void setUp()
+  {
+    def nodesCount = workers.size() + 1
+    if(nodeSshUtils.getNodeIds().size() == nodesCount) {
+      log.info("All nodemanagers are running..Skip restart")
+    } else {
+      restartNodeManagers()
+      retryUntil({
+        nodeSshUtils.getNodeIds().size() == nodesCount
+      }, MINUTES.toMillis(2))
+    }
+  }
+
+  def void restartNodeManagers ()
+  {
+    List<String> command = ['/etc/init.d/hadoop-yarn-nodemanager restart || true']
+    nodeSshUtils.runOnNode(master, command)
+    workers.each { String node ->
+      nodeSshUtils.runOnNode(node, command)
+    }
+    command = ['/etc/init.d/hadoop-yarn-resourcemanager restart || true']
```

Comment:
`|| true`?

Why it fails? Maybe try to fix this instead of ignoring, for example you can check status before restart.